contracts/CoverageDataProvider.sol

```diff
@@ -0,0 +1,172 @@
+        _underwritingPoolStatus[pools_[i]] = true;
+        emit UnderwritingPoolAdded(pools_[i]);
+      }
+      _underwritingPoolCount = index;
+    }
+
+    /**
+     * @notice Updates the status of the given underlying pool.
+     * @param pool_ The pool to update status.
+     * @param status_ The new pool status.
+    */
+    function setPoolStatus(address pool_, bool status_) external override onlyGovernance {
+      require(pool_ != address(0x0), "zero address pool");
+      require(_underwritingPoolToIndex[pool_] > 0, "invalid pool");
+      _underwritingPoolStatus[pool_] = status_;
+    }
+
+    /**
+     * @notice Sets the price oracle address.
+     * @param priceOracle_ The new price oracle address.
+    */
+    function setPriceOracle(address priceOracle_) external override onlyGovernance {
+      require(priceOracle_ != address(0x0), "zero address price oracle");
+      _priceOracle = IAavePriceOracle(priceOracle_);
+    }
+
+    /**
+     * @notice Returns the price oracle.
+     * @return oracle The address of the price oracle.
+    */
+    function priceOracle() public view override returns (address oracle) {
+      return address(_priceOracle);
+    }
+
+    /**
+     * @notice Returns status of the underwriting pool.
+     * @param pool_ The underwriting pool to query status.
+     * @return status True if underwriting pool is enabled. 
+    */
+    function poolStatus(address pool_) public view override returns (bool status) {
+      return _underwritingPoolStatus[pool_];
+    }
+
+    /**
+     * @notice Returns the underwriting pool for the given index.
+     * @param index_ The underwriting pool index.
+     * @return pool The underwriting pool.
+    */
+    function poolAt(uint256 index_) external view override returns (address pool) {
+      return _indexToUnderwritingPool[index_];
+    }
+
+    /**
+     * @notice Returns the underwriting pool count.
+     * @return count The number of underwriting pools.
+    */
+    function poolCount() public view override returns (uint256 count) {
+      return _underwritingPoolCount;
+    }
+
+    /**
+     * @notice The maximum amount of cover that Solace as a whole can sell.
+     * @return cover The max amount of cover in wei.
+    */
+    function maxCover() external view override returns (uint256 cover) {
+      // get vault balance
+      address vault = _registry.vault();
+      if (vault != address(0x0)) {
+        cover += IVault(payable(vault)).totalAssets();
+      }
+      // get pool balance
+      uint256 pools = poolCount();
+      for (uint256 i = 0; i < pools; i++) {
+        cover += getPoolBalance(_indexToUnderwritingPool[i]);
+      }
+    }
+
+    /**
+     * @notice Returns total value in `ETH` for the given underwriting pool.
+     * @param pool_ The underwriting pool.
+     * @return amount The total asset value of the underwriting pool in ETH.
+    */
+    function getPoolBalance(address pool_) public view override returns (uint256 amount) {
```

Comment:
imo it would feel more natural to handle each asset separately and not have to convert it all to eth. idk if its possible or what dependencies it would affect. not saying to change it but its something we should think about

Reply:
For now, I converted them to eth. But the note is taken.